server/src/api/roadmaps/roadmaps.controller.ts

```diff
@@ -96,3 +96,12 @@ export const deleteRoadmaps: RouteHandlerFnc = async (ctx) => {
 
   ctx.status = numDeleted == 1 ? 200 : 404;
 };
+
+export const leaveRoadmap: RouteHandlerFnc = async (ctx) => {
+  const numDeleted = await Role.query().deleteById([
+    Number(ctx.state.user!.id),
```

Comment:
I'd add undefined check that's used in many other routes as well
```suggestion
  if (!ctx.state.user) throw new Error('User is required');
  const numDeleted = await Role.query().deleteById([
    Number(ctx.state.user.id),
```